src/app.js

```diff
@@ -1,3 +1,39 @@
 'use strict';
 
-// Write your code here
+'use strict';
+
+const readline = require('readline');
+const { generateRandomNumber } = require('./generateRandomNumber');
+const { checkIsValidUserInput } = require('./checkIsValidUserInput');
+const { getBullsAndCows } = require('./getBullsAndCows');
+
+const terminal = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+function playGame() {
+  const secretNumber = generateRandomNumber().toString();
+
+  function askGuess() {
+    terminal.question('Enter your number: ', (userInput) => {
+      if (!checkIsValidUserInput(userInput)) {
+        return askGuess();
+      }
+
+      const { bulls } = getBullsAndCows(secretNumber, userInput);
+
+      if (bulls === 4) {
+        terminal.close();
```

Comment:
Consider providing a message to the user when they have guessed the correct number, indicating that they have won the game. This will improve user experience by clearly signaling the end of the game.